.github/workflows/compile.yml

```
@@ -19,66 +19,163 @@ jobs:
+          key: libvmi-${{ steps.get-hash.outputs.LIBVMI_HASH }}
+
+      - name: Build LibVMI
+        if: steps.cache-libvmi.outputs.cache-hit != 'true'
         run: |
+          rm -rfv libvmi
+          sudo apt-get install -y build-essential autoconf-archive flex bison libjson-c-dev libxen-dev debhelper
           git submodule update --init libvmi
           cd libvmi
-          autoreconf -vif
-          ./configure --disable-kvm --disable-file --disable-bareflank --disable-examples --disable-vmifs
-          make -j2
-          sudo make install
+          sed -i 's/--disable-kvm/--disable-kvm --disable-file --disable-bareflank --disable-examples --disable-vmifs/g' debian/rules
+          dpkg-buildpackage -B
+          mkdir dist
+          mv ../*.deb dist/
+
+      - name: Install LibVMI
+        run: |
+          cd libvmi/dist
+          sudo apt install -f ./*.deb
           sudo ldconfig
-          cd ..
+          cd ../..
+
+    outputs:
+      XEN_HASH: ${{ steps.get-hash.outputs.XEN_HASH }}
+      LIBVMI_HASH: ${{ steps.get-hash.outputs.LIBVMI_HASH }}
+
+  compile:
+    runs-on: ${{ matrix.os }}
+    needs:
+      - init
+    strategy:
+        matrix:
+          os:
+            - 'ubuntu-20.04'
+          flags:
+            - ''
+            - '--enable-debug'
+            - '--enable-debug --disable-plugin-syscalls'
+            - '--enable-debug --enable-sanitize'
+            - '--enable-debug --enable-repl'
+    env:
+      CC: clang
+      CXX: clang++
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          sudo apt-get update -q
+          sudo apt-get remove -y clang
+          sudo apt-get install -y clang-10 build-essential autotools-dev autoconf-archive flex bison libjson-c-dev liblzo2-dev libglib2.0-dev libtool
+          sudo pip3 install ctypesgen ipython
+
+      - name: Cache Xen debball
+        uses: actions/cache@v2
+        with:
+          path: xen/dist
+          key: xen-${{ needs.init.outputs.XEN_HASH }}
+
+      - name: Cache Libvmi files
+        uses: actions/cache@v2
+        with:
+          path: libvmi/dist
+          key: libvmi-${{ needs.init.outputs.LIBVMI_HASH }}
+
+      - name: Install Xen debball
+        run: |
+          sudo apt-get install -f ./xen/dist/xen-*.deb
+
+      - name: Install LibVMI
+        run: |
+          cd libvmi/dist
+          sudo apt install -f ./*.deb
+          sudo ldconfig
+          cd ../..
+
       - name: autoreconf
         run: autoreconf -vif
-      - name: Compile
-        run: |
-          ./configure
-          make -j2
-          make -j2 clean
-      - name: Compile debug
+
+      - name: Compile ${{ matrix.flags }}
         run: |
-          ./configure --enable-debug
-          make -j2
-          make -j2 clean
-      - name: Compile with plugin disabled
+          ./configure ${{ matrix.flags }}
+          make -j4
+
+  build:
+    runs-on: ${{ matrix.os }}
+    needs:
+      - compile
```

```suggestion
  build:
    runs-on: ${{ matrix.os }}
    needs:
      - init
```
@tklengyel what would you prefer btw, the tarball build depending on `compile` or `init` ?

---

init sounds fine